index.ts

```diff
@@ -271,19 +271,27 @@ export class PullFlatList<T> extends Component<PullFlatListProps<T>, State<T>> {
 
   public componentDidUpdate(prevProps: PullFlatListProps<T>) {
     const nextProps = this.props;
-    const nextGetReadable = nextProps.getScrollStream;
-    const prevGetReadable = prevProps.getScrollStream;
+    const { getScrollStream: nextGetScrollReadable, getPrefixStream: nextGetPrefixReadable } = nextProps;
+    const { getScrollStream: prevGetScrollReadable, getPrefixStream: prevGetPrefixReadable } = prevProps;
 
-    if (nextGetReadable !== prevGetReadable) {
+    if (nextGetScrollReadable !== prevGetScrollReadable) {
       this.stopScrollListener(() => {
         if (!this.unmounting) {
           this.isPulling = false;
           this.morePullQueue = 0;
           this.iteration = 0;
-          this.startScrollListener(nextGetReadable?.());
+          this.startScrollListener(nextGetScrollReadable?.());
         }
       });
     }
+
+    if (nextGetPrefixReadable !== prevGetPrefixReadable) {
+      this.stopPrefixListener(() => {
+        if (!this.unmounting) {
+          this.startPrefixListener(nextGetPrefixReadable?.());
+        }
```

Comment:
do any updates to stateful fields need to happen here, similar to the scroll listener code above?